ssl/s3_enc.c

```diff
@@ -466,8 +467,12 @@ size_t ssl3_final_finish_mac(SSL *s, const char *sender, size_t len,
 
     if (sender != NULL) {
         OSSL_PARAM digest_cmd_params[3];
+        int cmd;
+        void *msg;
+
+        ssl3_digest_master_key_set_params(s->session, &cmd,
+                                          &msg, digest_cmd_params);
 
-        ssl3_digest_master_key_set_params(s->session, digest_cmd_params);
         if (EVP_DigestUpdate(ctx, sender, len) <= 0
             || EVP_MD_CTX_set_params(ctx, digest_cmd_params) <= 0
```

Comment:
You're reverting to use `EVP_MD_CTX_ctrl()` in ssl/statem/statem_lib.c...  why not here?

Reply:
Because in statem_lib.c the operation is a DigestSign/DigestVerify which hasn't been updated to handle providers yet. This is a plain Digest operation which knows how to handle providers.

Reply:
Ah!